### PR TITLE
Live query scroll

### PR DESCRIPTION
**Pull request by avanbrunt-cb**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Tests have been added that prove the fix is effective or that the feature works.
- [x] New and existing tests pass locally with the changes.
- [x] Code follows the style guidelines of this project (PEP8, clean code, linter).
- [x] A self-review of the code has been done.

## What is the ticket or issue number?

CBAPI-5182

## Pull Request Description

Add scroll functionality for Live Query Result

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## How Has This Been Tested?

Manual and unit tests

### CODE REVIEW

File: src/tests/unit/audit_remediation/test_audit_remediation_base.py [hunk 1]
```diff
@@ -478,3 +480,72 @@ def test_run_async_faceting_query(cbcsdk_mock):
...
+    assert query._criteria["time_received"] == {
+        "start": "2023-12-10T00:00:00.000Z",
+        "end": "2023-12-11T00:00:00.000Z"
+    }
+
+    with pytest.raises(ApiError):
+        query.set_time_received(start="2023-12-10T00:00:00.000Z", end="2023-12-11T00:00:00.000Z", range="-3h")
+
+
+def test_result_scroll(cbcsdk_mock):
```
Comment (abowersox-cb) [inline src/tests/unit/audit_remediation/test_audit_remediation_base.py:508]: You have a unit test failure in this test.

Comment (avanbrunt-cb) [reply src/tests/unit/audit_remediation/test_audit_remediation_base.py:508]: Thanks! I forgot I modified that

File: docs/live-query.rst [hunk 2]
```diff
@@ -125,6 +125,34 @@ The sequence of calls are:
     >>> finished_job.get_output_as_file("/Users/myname/mydir/livequeryresults_async.csv")
 
 
+Scroll results
+--------------
+
+If you would like to ingest all the Live Query results whether that be from one Run or multiple Runs consider using the scroll option
+to fetch the latest results. The scroll option is limited to the last 24 hours for results across all Runs. You either need to specify
+a time_received or a list of one or more Run ids
```
Comment (abowersox-cb) [inline docs/live-query.rst:133]: This line should end with a :: (double colon) so the indented paragraph below will be displayed in monospace.

File: docs/live-query.rst [hunk 3]
```diff
@@ -125,6 +125,34 @@ The sequence of calls are:
...
+    >>> result_query = api.select(Result).set_time_received(range="-3h")
+    >>> list_results = result_query.scroll(10)
+    >>> print(f"num_remaining: {result_query.num_remaining}")
+    num_remaining: 35
+    >>> while result_query.num_remaining > 0:
+    >>>     list_results.extend(result_query.scroll(10))
+    >>> print(f"total results: len(list_results)")
+    total_results: 45
+
+Alternatively if you wanted to get all the results over multiple days for a single Run then use the Run's id
```
Comment (abowersox-cb) [inline docs/live-query.rst:144]: Same as above.

**Review by abowersox-cb — APPROVED** [review]
Other than a couple of doc fixes this is good.

File: docs/live-query.rst [hunk 4]
```diff
@@ -125,6 +125,34 @@ The sequence of calls are:
...
+to fetch the latest results. The scroll option is limited to the last 24 hours for results across all Runs. You either need to specify
+a time_received or a list of one or more Run ids
+
+    >>> result_query = api.select(Result).set_time_received(range="-3h")
+    >>> list_results = result_query.scroll(10)
+    >>> print(f"num_remaining: {result_query.num_remaining}")
+    num_remaining: 35
+    >>> while result_query.num_remaining > 0:
+    >>>     list_results.extend(result_query.scroll(10))
+    >>> print(f"total results: len(list_results)")
```
Comment (kebringer-cb) [inline docs/live-query.rst:141]: Missing {} around len(list_results).
Also at line 152.

**Review by kebringer-cb — COMMENTED** [review]
looks good. waiting for inline snippets.
guide was essential.

**Review by kebringer-cb — APPROVED** [review]

**Pull request merged**